source/features/conflict-marker.tsx

```diff
@@ -1,96 +1,65 @@
 import './conflict-marker.css';
 import React from 'dom-chef';
-import select from 'select-dom';
-import {AlertIcon} from '@primer/octicons-react';
-import oneMutation from 'one-mutation';
 import * as pageDetect from 'github-url-detection';
+import {AlertIcon} from '@primer/octicons-react';
+import batchedFunction from 'batched-function';
 
 import features from '../feature-manager.js';
 import api from '../github-helpers/api.js';
+import observe from '../helpers/selector-observer.js';
 
-type PRConfig = {
-	number: string;
-	user: string;
-	repo: string;
-	link: HTMLAnchorElement;
-	key: string;
-};
+async function addIcon(links: HTMLAnchorElement[]): Promise<void> {
+	const prConfigs = links.map(link => {
+		const [, owner, name, , prNumber] = link.pathname.split('/');
+		const key = api.escapeKey(owner, name, prNumber);
+		return {
+			key, link, owner, name, number: Number(prNumber),
+		};
+	});
 
-function createQueryFragment(pr: PRConfig): string {
-	return `
-		${pr.key}: repository(owner: "${pr.user}", name: "${pr.repo}") {
-			pullRequest(number: ${pr.number}) {
+	// Batch queries cannot be exported to .gql files
+	const batchQuery = prConfigs.map(({key, owner, name, number}) => `
+		${key}: repository(owner: "${owner}", name: "${name}") {
+			pullRequest(number: ${number}) {
 				mergeable
 			}
 		}
-	`;
-}
-
-function buildQuery(prs: PRConfig[]): string {
-	return prs.map(pr => createQueryFragment(pr)).join('\n');
-}
-
-function getPRConfig(prIcon: Element): PRConfig {
-	const link = prIcon.closest('.js-navigation-item')!.querySelector('a.js-navigation-open')!;
-	const [, user, repo, , number] = link.pathname.split('/');
-	return {
-		user,
-		repo,
-		number,
-		link,
-		key: api.escapeKey(user, repo, number),
-	};
-}
-
-async function init(): Promise<false | void> {
-	// Milestone issues are lazy-loaded
-	if (pageDetect.isMilestone()) {
-		// TODO: Use observe instead
-		await oneMutation(select('.js-milestone-issues-container')!, {childList: true});
-	}
-
-	const openPrIcons = select.all('.js-issue-row .octicon-git-pull-request.color-fg-open');
-	if (openPrIcons.length === 0) {
-		return false;
-	}
+	`).join('\n');
 
-	const prs = openPrIcons.map(icon => getPRConfig(icon));
-	const data = await api.v4(buildQuery(prs));
+	const data = await api.v4(batchQuery);
 
-	for (const pr of prs) {
+	for (const pr of prConfigs) {
 		if (data[pr.key].pullRequest.mergeable === 'CONFLICTING') {
 			pr.link.after(
 				<a
 					className="rgh-conflict-marker tooltipped tooltipped-e color-fg-muted ml-2"
 					aria-label="This PR has conflicts that must be resolved"
 					href={`${pr.link.pathname}#partial-pull-merging`}
 				>
-					<AlertIcon className="v-align-middle"/>
-				</a>,
+					<AlertIcon className="v-align-middle" />
+				</a>
 			);
 		}
 	}
 }
 
+function init(signal: AbortSignal): void {
+	observe('.js-issue-row:has(.octicon-git-pull-request.color-fg-open) a.js-navigation-open', batchedFunction(addIcon, {delay: 100}), {signal});
```

Comment:
Without a `delay` it's too fast even for high-speed internet:

<img width="266" alt="Screenshot" src="https://github.com/refined-github/refined-github/assets/1402241/a3357a10-2bd0-4fad-9a21-b0be02ba2f8f">

100ms seems to fix it:

<img width="266" alt="Screenshot 1" src="https://github.com/refined-github/refined-github/assets/1402241/26971e41-51b3-4542-b7cf-eb423314d066">



We have 2 more functions that use `batchedFunction` and likely suffer the same issue. PR welcome in that case too, inlining the `batchedFunction()` call this way